DSC/dsc.py

```diff
@@ -126,7 +131,60 @@ def get_distro_category():
     elif distro_name == 'suse':
         return DistroCategory.suse 
     waagent.AddExtensionEvent(name=ExtensionShortName, op='InstallInProgress', isSuccess=True, message="Unsupported distro :" + distro_name + "; distro_version: " + distro_version)
-    hutil.do_exit(51, 'Install', 'error', '51', distro_name + 'is not supported.')
+    hutil.do_exit(UnsupportedDistro, 'Install', 'error', str(UnsupportedDistro), distro_name + 'is not supported.')
+    
+def check_supported_OS():
+    """
+    Checks if the VM this extension is running on is supported by DSC
+    Returns for platform.linux_distribution() vary widely in format, such as
+    '7.3.1611' returned for a VM with CentOS 7, so the first provided
+    digits must match.
+    All other distros not supported will get error code 51
+    """
+    supported_dists = {'redhat' : ['6', '7'], # CentOS
+                       'centos' : ['6', '7'], # CentOS
+                       'red hat' : ['6', '7'], # Redhat
+                       'debian' : ['8'], # Debian
+                       'ubuntu' : ['14.04', '16.04', '18.04'], # Ubuntu
+                       'suse' : ['11', '12'], #SLES
+                       'opensuse' : ['13', '42.3'] #OpenSuse
+    }
+    vm_supported = False
+
+    try:
+        vm_dist, vm_ver, vm_id = platform.linux_distribution()
```

Comment:
This has been deprecated since version 3.5, will be removed in version 3.7. You will run into problems with distros that ship with python 3.7 or higher.
https://docs.python.org/3.5/library/platform.html#platform.linux_distribution

try parsing /etc/os-release and similar files like /etc/lsb-release  instead

Reply:
We currently do not support any distros with Python 3. We will take address this in later changes when support for Python 3 distros in added in future.